File: TME9/RouleTaBille/RouleTaBille/ViewController.m
//
//  ViewController.m
//  RouleTaBille
//
//  Created by Max Corp on 12/4/16.
//  Copyright © 2016 Max Corp. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    _backgroundImageArray = [[NSArray alloc] initWithObjects:@"rouletabille-fond-eau", @"rouletabille-fond-mosaique", @"rouletabille-fond-phobos", @"rouletabille-fond-voutes", nil];
    
    _indexImageArray = 0;
    
    self.view.backgroundColor = [UIColor colorWithPatternImage:[UIImage imageNamed:_backgroundImageArray[_indexImageArray]]];
    
    [self becomeFirstResponder];
    
    _startGameV = [[StartGameView alloc]init];
    _endGameV = [[EndGameView alloc] init];
    
    _startGameV.delegate = self;
    _endGameV.delegate = self;
    
    [self.view addSubview:_startGameV];
    [self.view addSubview:_endGameV];
    
    [_backgroundImageArray release];
    [_startGameV release];
    [_endGameV release];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)prefersStatusBarHidden{
    return YES;
}

/*
- (UIStatusBarStyle)preferredStatusBarStyle{
    return UIStatusBarStyleLightContent;
}*/

- (BOOL)shouldAutorotate{
    return NO;
}

/****** UIResponder ******/
- (void)motionBegan:(UIEventSubtype)motion withEvent:(UIEvent *)event{
    if (motion == UIEventSubtypeMotionShake){
        [self nextBackgroundImageOnShake];
    }
}

- (BOOL)canBecomeFirstResponder{
    return YES;
}

/****** Functions ******/
- (void)nextBackgroundImageOnShake{
    _indexImageArray = (_indexImageArray + 1) % _backgroundImageArray.count;
    self.view.backgroundColor = [UIColor colorWithPatternImage:[UIImage imageNamed:_backgroundImageArray[_indexImageArray]]];
}

- (void)startGame{
    
}

/****** Protocol StartGameViewDelegate ******/
- (void)shouldStartGame{
    [_startGameV hideView];
    [self startGame];
}

/****** Protocol EndGameViewDelegate ******/
- (void)displayWithScore:(int)score{
    
}

@end

File: TME7/iSouvenir/iSouvenir/ViewController.swift
//
//  ViewController.swift
//  iSouvenir
//
//  Created by m2sar on 24/11/2016.
//  Copyright © 2016 UPMC. All rights reserved.
//

import UIKit
import CoreLocation

class ViewController: UIViewController {
    var mV : myView!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        mV = myView(frame : self.view.frame)
        self.view = mV
        // Do any additional setup after loading the view, typically from a nib.
        
        if CLLocationManager.locationServicesEnabled(){
            self.view = mV
        }else{
            self.view = UIView()
            let alert = UIAlertController(title: "Erreur", message: "Localisation désactivée", preferredStyle: .actionSheet)
            alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil))
            self.present(alert, animated: true, completion: nil)
        }
        
        mV.photoPicker.sourceType = .photoLibrary
        present(mV.photoPicker, animated: true, completion: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

}

